### DISCUSSION TITLE
How to delete all images with low resolution?

### DISCUSSION THREAD

**Discussion by juxeii** [immich-app/immich#7702] [Q&A] [answered]
Is there a simple way to delete all images up to a certain resolution threshold?
Say I want to delete all photos with less than 500KB. How would I do that?

**Comment by mmomjian** [comment]
I don't think this is built-in. You can run the following command in Postgres to show all pictures < 500,000 bytes in your database.

`SELECT assets.id,CONCAT(ROUND(exif."fileSizeInByte" / 1048576.0::numeric , 2),'MB') AS fileSize FROM assets JOIN exif ON assets.id=exif."assetId" WHERE exif."fileSizeInByte" < 500000 ORDER BY exif."fileSizeInByte" ASC;`

To show the 100 smallest images:

`SELECT assets.id,CONCAT(ROUND(exif."fileSizeInByte" / 1048576.0::numeric , 2),'MB') AS fileSize FROM assets JOIN exif ON assets.id=exif."assetId" ORDER BY exif."fileSizeInByte" ASC LIMIT 100;`

You could then use the API `deleteAssets` endpoint to remove the matching pictures from the libraries.

**Comment by mmomjian** [comment, marked as answer]
I think a reasonable feature for the search dialog could be to specify either a minimum or maximum filesize to be compared to exif.fileSizeInBytes.
